specs/agents/tracing-instrumentation-db.md

```diff
@@ -141,6 +150,19 @@ The Elasticsearch cluster name is not always available in ES clients, as a resul
 - Instrument `_node/http` calls and cache the result in the agent with `host:port` as key.
 - execute a request to Elasticsearch and cache the result in the agent with `host:port` as key.
 
+#### `elasticsearch_capture_body_urls` configuration
+
+The URL patterns for which the agent is capturing the request body for Elasticsearch clients.
+ 
+Agents MAY offer this configuration option.
+If they don't, they MUST use a hard-coded list of URLs that correspond with the default value of this option.
+
+|                |            |
+|----------------|------------|
+| Type           | `List<`[`WildcardMatcher`](../../tests/agents/json-specs/wildcard_matcher_tests.json)`>` |
+| Default        | `*_search, *_msearch, *_msearch/template, *_search/template, *_count`                    |
```

Comment:
Some questions and thinking out loud on the default WildcardMatcher patterns. Please correct me where I go wrong or you have some experience or opinion.

Comparing to the current regex the Node.js agent uses for this selection:

```
// [...] We match
// a complete URL path component to attempt to avoid accidental matches of
// user data, like `GET /my_index_search/...`.
const pathIsAQuery = /\/(_search|_msearch|_count|_async_search|_sql|_eql)(\/|$)/
```

1. I think we can drop the ".../template" examples, because they will match the earlier patterns.

    ```
    *_search, *_msearch, *_count
    ```

2. To clarify, the proposal is to match against the full URL path (e.g. `/my-data-stream/_eql/search`), right? If so, we'll need a trailing wildcard.

    ```
    *_search*, *_msearch*, *_count*
    ```

3. Would it make sense to have the default patterns include the leading '/' to avoid accidentally capturing the body for something like `POST /my_counterintelligence_dump/_bulk`? 

    ```
    */_search*, */_msearch*, */_count*
    ```

    API paths commonly do *not* have the trailing slash and with WildcardMatcher patterns we cannot match against "a slash or end of string" like the `(\/|$)` regex segment above. That's probably fine as an accidental match would require a user index or data stream to start with an underscore.

4. The existing Node.js regex includes a few others as well:

    ```
    */_search*, */_msearch*, */_count*, */_async_search*, */_sql*, */_eql*
    ```

5. Do we want to start with including a number of other "search-y" Elasticsearch REST APIs? I don't know Elasticsearch well at all, so I am happy to defer to others' experience. Potential options:

    - `/_knn_search` https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search-api.html
    - `/_terms_enum` https://www.elastic.co/guide/en/elasticsearch/reference/current/search-terms-enum.html
    - `/_explain` https://www.elastic.co/guide/en/elasticsearch/reference/current/search-explain.html
    - `/_rank_eval` https://www.elastic.co/guide/en/elasticsearch/reference/current/search-rank-eval.html
    - `/_validate` https://www.elastic.co/guide/en/elasticsearch/reference/current/search-validate.html
    - `/_render` https://www.elastic.co/guide/en/elasticsearch/reference/current/render-search-template-api.html
    - `/_mvt` https://www.elastic.co/guide/en/elasticsearch/reference/current/search-vector-tile-api.html
    - `/_delete_by_query` https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html
    - `/_update_by_query` https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html
    - `/_graph` https://www.elastic.co/guide/en/elasticsearch/reference/current/graph-explore-api.html

Reply:
1, 2) I think adding a trailing wildcard could lead to us capturing things that should be captured. One (bad) example is [`DELETE /_eql/search/<search_id>`](https://www.elastic.co/guide/en/elasticsearch/reference/current/delete-async-eql-search-api.html). Bad example because there isn't actually a request body to be captured for this API.

3) Good point. Although, if there's no trailing wildcard, the chances are relatively slim. But a leading `/` doesn't hurt.

> That's probably fine as an accidental match would require a user index or data stream to start with an underscore.

ES disallows the first character to start with an underscore

4) Let's remove the trailing wildcards:

```
*/_search, */_msearch, */_count, */_async_search, */_sql, */_eql/search
```

5) Maybe some of these. But definitely not `/_mvt` (has no body) and `/_update_by_query` (may capture user data). But either way, I think adding more APIs by default than we do now is orthogonal to this PR.

Reply:
```suggestion
| Default        | `*/_search, */_search/template, */_msearch, */_msearch/template, */_async_search, */_count, */_sql, */_eql/search`|
```

Reply:
Thanks for the clarification. I was missing a basic point on not needing/wanting the trailing wildcard. (A trailing '/' *is* allowed by ES, e.g. `GET /my-index/_search/`, but the clients we are instrumenting likely don't add a trailing slash. The JS client doesn't.)

I guess the leading '/' in the match patterns (i.e. `*/_search` rather than `*_search`) is slightly helpful to avoid overlap with other ES APIs (like `/_knn_search` in this example).  So your current suggestion looks good to me.